clippy_lints/src/matches.rs

```diff
@@ -713,16 +713,18 @@ fn all_ranges<'a, 'tcx>(cx: &LateContext<'a, 'tcx>, arms: &'tcx [Arm<'_>]) -> Ve
             } = *arm
             {
                 if let PatKind::Range(ref lhs, ref rhs, ref range_end) = pat.kind {
-                    let lhs = constant(cx, cx.tables, lhs)?.0;
-                    let rhs = constant(cx, cx.tables, rhs)?.0;
-                    let rhs = match *range_end {
-                        RangeEnd::Included => Bound::Included(rhs),
-                        RangeEnd::Excluded => Bound::Excluded(rhs),
-                    };
-                    return Some(SpannedRange {
-                        span: pat.span,
-                        node: (lhs, rhs),
-                    });
+                    if let (Some(l), Some(r)) = (lhs, rhs) {
```

Comment:
(Leaving this comment as a motivational example for https://github.com/rust-lang/rust/issues/65490.)

Reply:
Also, is there a reason not to support `None` at each end here?

Reply:
I'm not sure how to pass the `Expr`s to `constant()` on half patterns, any thoughts?

Reply:
This is what the compiler does: https://github.com/rust-lang/rust/pull/67258/files#diff-e826542d6f24137d5b0b5ae333230529R493-R511